core/src/main/java/tech/tablesaw/io/ReadOptions.java

```diff
@@ -327,8 +357,112 @@ public Builder minimizeColumnSizes() {
       return this;
     }
 
+    /**
+     * Provide column types for all columns preventing autodetect column type logic. It's expected
+     * that the array contains all columns
```

Comment:
"The array must contain a ColumnType for each column in the table" would be more accurate. An error will be thrown if they don't match up.